bundle/Templating/Twig/Extension/ContentViewRuntime.php

```diff
@@ -127,29 +139,40 @@ private function viewMatched(ContentView $contentView): bool
         return !empty($contentView->getConfigHash());
     }
 
-    private function getContent(ValueObject $contentOrLocation): Content
+    private function getContent(ValueObject $contentOrLocation): ValueObject
     {
-        if ($contentOrLocation instanceof Content) {
+        if ($contentOrLocation instanceof Content || $contentOrLocation instanceof APIContent) {
             return $contentOrLocation;
         }
 
-        if ($contentOrLocation instanceof Location) {
+        if ($contentOrLocation instanceof Location || $contentOrLocation instanceof APILocation) {
+            // eZ location also has a lazy loaded "content" property
             return $contentOrLocation->content;
         }
 
         throw new LogicException('Given value must be Content or Location instance.');
     }
 
-    private function getLocation(ValueObject $contentOrLocation): ?Location
+    private function getLocation(ValueObject $contentOrLocation): ?ValueObject
     {
-        if ($contentOrLocation instanceof Location) {
+        if ($contentOrLocation instanceof Location || $contentOrLocation instanceof APILocation) {
             return $contentOrLocation;
         }
 
         if ($contentOrLocation instanceof Content) {
             return $contentOrLocation->mainLocation;
         }
 
+        if ($contentOrLocation instanceof APIContent) {
+            // View builder will not load the main location if it is not provided,
+            // this makes sure it is available to the template
+            try {
+                return $this->locationService->loadLocation($contentOrLocation->contentInfo->mainLocationId);
```

Comment:
This will be a problem if main Location does not exist and the ID parameter is hinted as integer.

Reply:
Right. Fixed.

I've removed the `try...catch` now since non-existent location in case content references it via main location ID is now a separate issue, pointing to data inconsistencies.